packages/cyberstorm/src/components/MultiSelectSearch/MultiSelectSearch.tsx

```diff
@@ -0,0 +1,205 @@
+      document.addEventListener("touchstart", closeOpenMenu);
+      return () => {
+        document.removeEventListener("mousedown", closeOpenMenu);
+        document.removeEventListener("touchstart", closeOpenMenu);
+      };
+    });
+
+    React.useEffect(() => {
+      onChange(selected);
+      const updatedOptions = options.filter(
+        (option) => !selected.includes(option) && option.label.includes(search)
+      );
+      setFilteredOptions(updatedOptions);
+    }, [selected, search]);
+
+    return (
+      <div className={styles.root} ref={ref}>
+        <div className={styles.selected}>
+          {selected.map((option, key) => {
+            return (
+              <Button.Root
+                key={key}
+                onClick={() => remove(option)}
+                colorScheme="default"
+                paddingSize="small"
+                style={{ gap: "0.5rem" }}
+              >
+                <Button.ButtonLabel>{option.label}</Button.ButtonLabel>
+                <Button.ButtonIcon>
+                  <FontAwesomeIcon icon={faXmark} />
+                </Button.ButtonIcon>
+              </Button.Root>
+            );
+          })}
+        </div>
+        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
+        <div
+          className={styles.search}
+          onClick={(e) =>
+            handleParentClick(e, () => {
+              inputRef.current && inputRef.current.focus();
+            })
+          }
+          role="button"
+          tabIndex={0}
+          ref={menuRef}
+        >
+          <div className={styles.inputContainer} data-color={color}>
+            <input
+              className={styles.input}
+              value={search}
+              data-color={color}
+              onFocus={() => setIsVisible(true)}
+              onChange={(e) => setSearch(e.currentTarget.value)}
+              ref={inputRef}
+              placeholder={placeholder}
+            />
+            <button
+              onClick={(e) => handleParentClick(e, () => setList([]))}
+              className={styles.removeAllButton}
+            >
+              <Icon inline>
+                <FontAwesomeIcon icon={faCircleXmark} />
+              </Icon>
+            </button>
+            <div className={styles.inputButtonDivider}></div>
+            <button
+              onClick={(e) =>
+                handleParentClick(e, () => setIsVisible(!isVisible))
+              }
+              className={styles.openMenuButton}
+            >
+              <Icon inline>
+                <FontAwesomeIcon icon={faCaretDown} />
+              </Icon>
+            </button>
+          </div>
+          <div
+            className={classnames(
+              styles.menu,
+              isVisible ? styles.visible : null
+            )}
+          >
+            {filteredOptions.map((option, key) => {
+              return (
+                <MultiSelectItem
+                  key={key}
```

Comment:
Ideally [the index shouldn't be used as the key](https://react.dev/learn/rendering-lists#why-does-react-need-keys). If we can guarantee that `option.value` can be unique within this `.map`, it would make a nice key.